docs/dev/e2eTesting-GettingStarted.md

```diff
@@ -0,0 +1,96 @@
+# E2E Testing
+## Getting Started
+
+This document describes the steps that need to be taken to run a e2e test from your machine that points to a 
+pre-configured network with a mirror-node.
+
+
+## Setup
+
+Create a configuration file `application.yaml` that has the node configuration, publisher configuration and 
+subscriber configuration along with an operator id and its key.
+
+Example :
+```
+hedera:
+  mirror:
+    monitor:
+      network: PREVIEWNET
+      operator:
+        accountId: 0.0.12345
+        privateKey: <privateKey>
+      publish:
+        clients: 4
+        threads: 40
+        warmupPeriod: 120s
+        scenarios:
+          - name: CryptoTransfer
+            duration: 5m
+            properties:
+              senderAccountId: 0.0.12345
+              recipientAccountId: 0.0.12346
+              amount: 1
+              transferType: CRYPTO
+            tps: 100
+            type: CRYPTO_TRANSFER
+      subscribe:
+        clients: 4
+        grpc:
+        rest:
+          - name: CryptoTransferSub
+            enabled: true
+            samplePercent: 0.05
+```
+
+> The above config targets the `PREVIEWNET` by sending cryptoTransfer transactions for 5 mins with 100 tps
```

Comment:
Is the TPS network target or each client tries to submit 100 TPS?

Reply:
@Neeharika-Sompalli, good catch. The doc should state clearly whether it is a total TPS or a TPS `per client` (or even per thread, since there is a number of threads setting here, namely `threads: 40`).

Reply:
We are specifying the network target TPS for that specific action. 
Let me know if the following makes sense :

In the config, I specify 40 publisher threads with 4 clients to perform the mentioned scenarios. Each scenario will have a network target TPS. The monitor will distribute these actions mentioned in the scenarios among the 4 clients which intern use the 40 publisher thread to send traffic according to the target TPS we specified.

@steven-sheehy  is this accurate ?

Also please read this text quoted from [here](https://github.com/hashgraph/hedera-mirror-node/blob/master/docs/monitor.md).

"For example, to achieve 10K TPS we set the number of clients to 4, and the number of threads to 40. Please adjust accordingly per your needs. The transaction publisher will round-robin the list of clients to send transactions to the Hedera Network, and every transaction is sent to a randomly chosen node, ensuring the load is distributed evenly across all nodes."

Reply:
Anirudh is correct. My answer to Leo from above:

> No, it's 100 TPS total. The clients and threads are just pools that the scenarios use. There's not dedicated threads per client since we're using asynchronous, non-blocking publishers (the SDK itself).

Reply:
I see. The TPS is for total. For example, if I want to have 10K TPS, I need to set the `tps: 10000`. The publisher threads number and clients number aren't directly related to the total TPS, but they need to be set to appropriate values so the monitoring system can reach the target TPS easily.

Reply:
Correct. The threads property is actually the response handling thread and not a publishing thread. We should probably rename it.

Reply:
We've renamed it to `responseThreads` in master.